examples/kepler-integration/src/components/export-video/export-video-panel-container.js

```diff
@@ -161,12 +139,54 @@ export class ExportVideoPanelContainer extends Component {
     });
     // setFileNameDeckAdapter(name.target.value);
   }
-  /* setQuality(resolution){
+  setQuality(resolution) {
     this.setState({
       quality: resolution
     });
-    setResolution(resolution);
-  }*/
+    if (resolution === 'Good (540p)') {
+      const newWidth = 960;
+      const newHeight = 540;
+      this.setState({
+        encoderSettings: {
+          ...this.state.encoderSettings,
+          gif: {
+            ...this.state.encoderSettings.gif,
+            width: newWidth,
+            height: newHeight
+          }
+          // TODO Add other encoders as needed. Not yet implemented
+        }
+      });
+    } else if (resolution === 'High (720p)') {
+      const newWidth = 1280;
+      const newHeight = 720;
+      this.setState({
+        encoderSettings: {
+          ...this.state.encoderSettings,
```

Comment:
This can be shorter. Imagine if there was just one setState at the bottom of this function, and `let newWidth` is initialized with the current width and then just updated in the if statement if it needs to be

Reply:
is this what you had in mind?

```javascript
    if (resolution === 'Good (540p)') {
      const newWidth = 960;
      const newHeight = 540;
    }
   // Do for other resolution options

    this.setState({
      quality: resolution,
      encoderSettings: {
        ...encoderSettings,
        gif: {
          ...encoderSettings.gif,
          width: newWidth,
          height: newHeight
        }
        // TODO Add other encoders as needed. Not yet implemented
      }
    });
```

Reply:
Yeah, very close. You will have to define the newWidth above the if statements though and as a `let`. The initial newWidth value is `this.state.encoderSettings.gif.width`

Reply:
yup learned that the hard way after struggling with scope for the past 15 or so mins. TIL const is block scope